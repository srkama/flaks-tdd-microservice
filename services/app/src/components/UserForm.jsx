import React from 'react';
import { Redirect } from 'react-router-dom';

const UserForm = (props) => {
    if (props.isAuthenticated) {
        return <Redirect to="/" />
    }
    return (
        <div>
            <h1>{props.title}</h1>
            <hr /><br />
            <form onSubmit={props.handleSubmit}>
                <div className="field">
                    <label className="label">Username</label>
                    <div className="control">
                        <input
                            name="username"
                            className="input"
                            type="text"
                            placeholder="Name"
                            value={props.formData.username}
                            onChange={props.handleChange} />
                    </div>
                </div>
                {props.type === 'registration' &&
                    <div className="field">
                        <label className="label">Email</label>
                        <div className="control">
                            <input
                                name="email"
                                className="input"
                                type="text"
                                placeholder="email"
                                value={props.formData.email}
                                onChange={props.handleChange} />
                        </div>
                    </div>
                }
                <div className="field">
                    <label className="label">Password</label>
                    <div className="control">
                        <input
                            name="password"
                            className="input"
                            type="password"
                            placeholder="password"
                            value={props.formData.password}
                            onChange={props.handleChange} />
                    </div>
                </div>
                <div className="field is-grouped">
                    <div className="control">
                        <input type="submit" name="submit" className="button is-link" value="Submit" />
                    </div>
                </div>
            </form >
        </div>
    )
}

export default UserForm
