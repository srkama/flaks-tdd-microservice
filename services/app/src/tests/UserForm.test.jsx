import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import UserForm from '../components/UserForm';


const formData = {
    username: '',
    email: '',
    password: ''
}
test('Registration Form renders without crashing', () => {
    const wrapper = shallow(<UserForm type='registration' title='Registration Form' formData={formData} />);
    const h1 = wrapper.find('h1');
    expect(h1.length).toBe(1);
    expect(h1.get(0).props.children).toBe('Registration Form');
    const formGroup = wrapper.find('.field').find('input');
    expect(formGroup.length).toBe(4);
    expect(formGroup.get(0).props.name).toBe('username');
    expect(formGroup.get(0).props.value).toBe('');
    expect(formGroup.get(0).props.type).toBe('text');
    expect(formGroup.get(1).props.name).toBe('email');
    expect(formGroup.get(1).props.value).toBe('');
    expect(formGroup.get(1).props.type).toBe('text');
    expect(formGroup.get(2).props.name).toBe('password');
    expect(formGroup.get(2).props.value).toBe('');
    expect(formGroup.get(2).props.type).toBe('password');
});


test('Login Form renders without crashing', () => {
    const wrapper = shallow(<UserForm type='login' title='Login Form' formData={formData} />);
    const h1 = wrapper.find('h1');
    expect(h1.length).toBe(1);
    expect(h1.get(0).props.children).toBe('Login Form');
    const formGroup = wrapper.find('.field').find('input');
    expect(formGroup.length).toBe(3);
    expect(formGroup.get(0).props.name).toBe('username');
    expect(formGroup.get(0).props.value).toBe('');
    expect(formGroup.get(0).props.type).toBe('text');
    expect(formGroup.get(1).props.name).toBe('password');
    expect(formGroup.get(1).props.value).toBe('');
    expect(formGroup.get(1).props.type).toBe('password');
});

test('Registration snapshot matches properly', () => {
    const tree = <UserForm type='registratoin' title='Registration Form' formData={formData} />
    expect(tree).toMatchSnapshot();
});

test('Login snapshot matches properly', () => {
    const tree = <UserForm type='login' title='Login Form' formData={formData} />
    expect(tree).toMatchSnapshot();
});