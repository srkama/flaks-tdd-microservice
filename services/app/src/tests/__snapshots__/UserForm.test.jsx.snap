// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot matches properly 1`] = `
<UserForm
  formData={
    Object {
      "email": "",
      "password": "",
      "username": "",
    }
  }
  title="Login Form"
  type="login"
/>
`;

exports[`Registration snapshot matches properly 1`] = `
<UserForm
  formData={
    Object {
      "email": "",
      "password": "",
      "username": "",
    }
  }
  title="Registration Form"
  type="registratoin"
/>
`;
